LabProject/src/main/java/agh/ics/oop/GrassField.java

```diff
@@ -0,0 +1,64 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.lang.Math;
+
+public class GrassField extends AbstractWorldMap{
+    private final int amountOfGrasses;
+
+    public GrassField(int amountOfGrasses){
+        this.amountOfGrasses=amountOfGrasses;
+        for (int  i=0; i<amountOfGrasses;i++){seedRandomGrass();}
+    }
+
+    public void seedRandomGrass(){
+        Random rand = new Random();
+        int x=rand.nextInt((int) Math.sqrt(amountOfGrasses*10)+1);
+        int y=rand.nextInt((int) Math.sqrt(amountOfGrasses*10)+1);
+        Vector2d newGrassPosition=new Vector2d(x,y);
+        if (!isOccupied(newGrassPosition)){
```

Comment:
jeśli pozycja będzie zajęta to w efekcie wylosujemy mniej niż `amountOfGrasses`... powinniśmy zapewnić żeby losowanie tworzyło tyle elementów, ile sobie zażyczyliśmy.